### PR TITLE
Add option +IOs to disable scheduler thread polling

### PR DESCRIPTION
**Pull request by sverker**
Default is true, scheduler thread polling enabled as before.

Scheduler polling has in some cases seen to cause longer read event latency, and can now be disabled at the expense of some increased cpu load.

### CODE REVIEW

File: erts/emulator/sys/common/erl_check_io.c [hunk 1]
```diff
@@ -93,24 +93,27 @@ typedef enum {
 #endif
     ERTS_EV_FLAG_WANT_ERROR    = 0x10,  /* ERL_NIF_SELECT_ERROR turned on */
 
-    /* Combinations */
-    ERTS_EV_FLAG_USED_FALLBACK = ERTS_EV_FLAG_USED | ERTS_EV_FLAG_FALLBACK,
-    ERTS_EV_FLAG_USED_SCHEDULER = ERTS_EV_FLAG_USED | ERTS_EV_FLAG_SCHEDULER,
-    ERTS_EV_FLAG_USED_IN_SCHEDULER = ERTS_EV_FLAG_USED | ERTS_EV_FLAG_SCHEDULER | ERTS_EV_FLAG_IN_SCHEDULER,
-    ERTS_EV_FLAG_UNUSED_SCHEDULER = ERTS_EV_FLAG_SCHEDULER,
-    ERTS_EV_FLAG_UNUSED_IN_SCHEDULER = ERTS_EV_FLAG_SCHEDULER | ERTS_EV_FLAG_IN_SCHEDULER
```
Comment (garazdawi) [inline erts/emulator/sys/common/erl_check_io.c:101]: The nice thing about this enum was that gdb manages to print the flag combinations.

File: erts/emulator/sys/common/erl_check_io.h [hunk 2]
```diff
@@ -112,6 +112,22 @@ typedef struct {
 } ErtsIoTask;
 
 
+ERTS_GLB_INLINE int erts_sched_poll_enabled(void);
+
+#if ERTS_GLB_INLINE_INCL_FUNC_DEF
+
+ERTS_GLB_INLINE int erts_sched_poll_enabled(void)
+{
+#if ERTS_POLL_USE_SCHEDULER_POLLING
```
Comment (garazdawi) [inline erts/emulator/sys/common/erl_check_io.h:121]: Should we rename this define to `ERTS_POLL_ENABLE_SCHEDULER_POLLING` or something similar? I was confused at first as to why the define was needed at all.

**Review by garazdawi — COMMENTED** [review]
In emulator there is the testcase scheduler_SUITE:poll_thread. I think we should add a testcase there that the option works.

Also, we ignore the option if ERTS_POLL_USE_SCHEDULER_POLLING is false, is that intentional?

**Comment by sverker** [comment]
> we ignore the option if ERTS_POLL_USE_SCHEDULER_POLLING is false, is that intentional?

The easy way out I guess. I have pushed a commit 5e7aa243a031c0804572740c6e8c84345a014f6d that refuse +IOs true if ERTS_POLL_USE_SCHEDULER_POLLING is false.

**Comment by sverker** [comment]
@garazdawi Three new commits, can you have a new look?

**Review by garazdawi — APPROVED** [review]
Looks good to me!

**Pull request merged**